crates/matrix-sdk-ui/src/timeline/pagination.rs

```diff
@@ -14,13 +14,36 @@
 
 use std::{fmt, ops::ControlFlow, sync::Arc, time::Duration};
 
-use eyeball::Subscriber;
+use eyeball::{SharedObservable, Subscriber};
 use matrix_sdk::event_cache::{self, BackPaginationOutcome};
 use tracing::{instrument, trace, warn};
 
 use super::Error;
 use crate::timeline::{event_item::RemoteEventOrigin, inner::TimelineEnd};
 
+struct ResetStatusGuard {
+    status: SharedObservable<PaginationStatus>,
+    target: Option<PaginationStatus>,
+}
+
+impl ResetStatusGuard {
+    fn new(status: SharedObservable<PaginationStatus>, target: PaginationStatus) -> Self {
```

Comment:
Any reason why target is the first argument here: https://github.com/matrix-org/matrix-rust-sdk/pull/3359/files#diff-a6ab59f890b91b82d1648c9c8bbd78c6387ece6d89f8a30683fe941be504628eR147 and the second in this code?

If not, I'd recommend keeping the order consistent.

Reply:
There's no reason, so you're right, we should make it consistent. I slightly prefer state first, then target, so I'll switch the other one around. Thanks!